src/modules/SimpleSlippageRouter.sol

```diff
@@ -0,0 +1,145 @@
+    using SafeTransferLib for ERC20;
+    using Math for uint256;
+    using Address for address;
+
+    /**
+     * @notice attempted to carry out tx with expired deadline.
+     * @param deadline specified tx block.timestamp to not pass during tx.
+     */
+    error SimpleSlippageAdaptor__ExpiredDeadline(uint64 deadline);
+
+    /**
+     * @notice attempted to carry out deposit() tx with less than acceptable minimum shares.
+     * @param minimumShares specified acceptable minimum shares amount.
+     * @param actualSharesQuoted actual amount of shares to come from proposed tx.
+     */
+    error SimpleSlippageAdaptor__DepositMinimumSharesUnmet(uint256 minimumShares, uint256 actualSharesQuoted);
+
+    /**
+     * @notice attempted to carry out withdraw() tx where required shares to redeem > maxShares specified by user.
+     * @param maxShares specified acceptable max shares amount.
+     * @param actualSharesQuoted actual amount of shares to come from proposed tx.
+     */
+    error SimpleSlippageAdaptor__WithdrawMaxSharesSurpassed(uint256 maxShares, uint256 actualSharesQuoted);
+
+    /**
+     * @notice attempted to carry out mint() tx where resultant required assets for requested shares is too much.
+     * @param minShares specified acceptable min shares amount.
+     * @param maxAssets specified max assets to spend on mint.
+     * @param actualAssetsQuoted actual amount of assets to come from proposed tx, indicating asset amount not enough for specified shares.
+     */
+    error SimpleSlippageAdaptor__MintMaxAssetsRqdSurpassed(
+        uint256 minShares,
+        uint256 maxAssets,
+        uint256 actualAssetsQuoted
+    );
+
+    /**
+     * @notice attempted to carry out redeem() tx where assets returned (the result of redeeming maxShares) < minimumAssets.
+     * @param maxShares specified acceptable max shares amount.
+     * @param minimumAssets specified minimum amount of assets to be returned.
+     * @param actualAssetsQuoted actual amount of assets to come from proposed tx.
+     */
+    error SimpleSlippageAdaptor__RedeemMinAssetsUnmet(
+        uint256 maxShares,
+        uint256 minimumAssets,
+        uint256 actualAssetsQuoted
+    );
+
+    /**
+     * @notice deposits assets into specified cellar w/ _minimumShares expected and _deadline specified.
+     * @param _cellar specified cellar to deposit assets into.
+     * @param _assets amount of cellar base assets to deposit.
+     * @param _minimumShares amount of shares required at min from tx.
+     * @param _deadline block.timestamp that tx must be carried out by.
+     */
+    function deposit(Cellar _cellar, uint256 _assets, uint256 _minimumShares, uint64 _deadline) public {
+        if (block.timestamp > _deadline) revert SimpleSlippageAdaptor__ExpiredDeadline(_deadline);
+        ERC20 baseAsset = _cellar.asset();
+        baseAsset.safeTransferFrom(msg.sender, address(this), _assets);
+        uint256 shares = _cellar.previewDeposit(_assets);
```

Comment:
Could you move this line, and the if statement below, to be right below the deadline revert? that way if we are going to revert, we revert earlier in the call before we transfer the users assets into the router

Reply:
Good point, done